examples/config/read-json.cpp

```diff
@@ -0,0 +1,87 @@
+// Illustrates how to read custom data types from JSON files.
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/json_reader.hpp"
+#include "caf/type_id.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <string_view>
+
+constexpr std::string_view example_input = R"([
+  {
+    "id": 1,
+    "name": "John Doe"
+  },
+  {
+    "id": 2,
+    "name": "Jane Doe",
+    "email": "jane@doe.com"
+  }
```

Comment:
I think JSON has a `null` value as well. Does that work equivalently to omitting the key-value pair?

Reply:
It should. Let me try and I'll file a new PR with a fix in case it doesn't. :)